### PR TITLE
NBS-4684: [Disk Manager] use proxy overlay disks in relocation if possible

### PR DESCRIPTION
**Pull request by BarkovBG**
need to use proxy overlay disk in relocation 

now data path looks like
snapshot -> nbs control -> host with volume tablet -> nbs control -> snapshot 

after this optimization it will be
snapshot -> nbs control -> snapshot

### CODE REVIEW

**Comment by BarkovBG** [comment]
тут надо large тесты прогнать

**Comment by BarkovBG** [comment]
> тут надо large тесты прогнать

все ок 
https://github.com/ydb-platform/nbs/actions/runs/7813679154

File: cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go [hunk 1]
```diff
@@ -119,6 +119,58 @@ func (t *replicateDiskTask) GetResponse() proto.Message {
...
+	created, err := client.CreateProxyOverlayDisk(
+		ctx,
+		proxyOverlayDiskID,
+		diskID, // baseDiskID
+		currentCheckpointID,
+	)
+	if err != nil {
+		return "", err
+	}
+
```
Comment (jkuradobery) [inline cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go:149]: Don't you need to save the state? What about if the task gets terminated by context?

Comment (BarkovBG) [reply cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go:149]: id does not metter

CreateVolume transaction is idempotent in nbs

**Review by jkuradobery — DISMISSED** [review]

File: cloud/disk_manager/internal/pkg/dataplane/protos/replicate_disk_task.proto [hunk 2]
```diff
@@ -39,6 +39,8 @@ message ReplicateDiskTaskState {
     // Store SecondsRemaining each UselessReplicationIterationsBeforeAbort.
     // If SecondsRemaining degrades, abort replication.
     int64 MeasuredSecondsRemaining = 9;
+
+    string ProxyOverlayDiskId = 10;
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/dataplane/protos/replicate_disk_task.proto:43]: No need for saving this id

File: cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go [hunk 3]
```diff
@@ -119,6 +127,61 @@ func (t *replicateDiskTask) GetResponse() proto.Message {
...
+	_, currentCheckpointID, _ := t.getCheckpointIDs(execCtx)
+	diskID := t.request.SrcDisk.DiskId
+	proxyOverlayDiskID := common.GetProxyOverlayDiskID(
+		diskID,
+		currentCheckpointID,
+	)
+
+	if len(proxyOverlayDiskID) == 0 {
+		return nil
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go:178]: Is this possible?

File: cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go [hunk 4]
```diff
@@ -119,6 +127,61 @@ func (t *replicateDiskTask) GetResponse() proto.Message {
...
+	)
+	if err != nil {
+		return "", err
+	}
+
+	if created {
+		return proxyOverlayDiskID, nil
+	}
+
+	return "", nil
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go:157]: Is this possible? What happens in this case?

File: cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go (hunk 3 above)
Comment (BarkovBG) [reply cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go:178]: https://github.com/ydb-platform/nbs/blob/8e8f5d5b4b226051f356dd955b05f36d4b28de0e/cloud/disk_manager/internal/pkg/clients/nbs/client.go#L608

File: cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go (hunk 4 above)
Comment (BarkovBG) [reply cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go:157]: https://github.com/ydb-platform/nbs/blob/8e8f5d5b4b226051f356dd955b05f36d4b28de0e/cloud/disk_manager/internal/pkg/clients/nbs/client.go#L608

Comment (BarkovBG) [reply cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go:157]: everything will be ok:
https://github.com/ydb-platform/nbs/blob/main/cloud/disk_manager/internal/pkg/dataplane/nbs/source.go#L249

**Review by jkuradobery — APPROVED** [review]

**Review by SvartMetal — APPROVED** [review]

**Review by gy2411 — APPROVED** [review]

**Pull request merged**